tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php

```diff
@@ -0,0 +1,196 @@
+        $this->currentMock->method('getResponse')
+            ->willReturn($response);
+
+        $expected = [
+            'street_address1' => 'Test street 123',
+            'street_address2' => 'Additional street 007',
+            'street_address3' => null,
+            'street_address4' => null,
+            'first_name'      => 'TesterFn',
+            'last_name'       => 'TesterLn',
+            'locality'        => 'Los Angeles',
+            'region'          => 'CA',
+            'postal_code'     => '10011',
+            'country_code'    => 'US',
+            'phone'           => '+1 123 111 3333',
+            'phone_number'    => '+1 123 111 3333',
+        ];
+
+        $this->currentMock->method('prepareAddressData')
+            ->willReturn($expected);
+
+
+        $this->currentMock->loadBlockAction();
+        $result = Mage::getSingleton('admin/session')->getOrderShippingAddress();
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testPrepareAddressData()
+    {
+        $actual = [
+            'street_address1' => 'Test street 123',
+            'street_address2' => 'Additional street 007',
+            'street_address3' => null,
+            'street_address4' => null,
+            'first_name'      => 'TesterFn',
+            'last_name'       => 'TesterLn',
+            'locality'        => 'Los Angeles',
+            'region'          => 'CA',
+            'postal_code'     => '10011',
+            'country_code'    => 'US',
+            'phone'           => '+1 123 111 3333',
+            'phone_number'    => '+1 123 111 3333',
+        ];
+
+        $postData = [
+            'shipping_address' => [
+                'street' => [
+                    'Test street 123',
+                    'Additional street 007',
+                ],
+                'firstname'     => 'TesterFn',
+                'lastname'      => 'TesterLn',
+                'city'          => 'Los Angeles',
+                'region_id'     => 12, // Region: CA
+                'postcode'      => '10011',
+                'country_id'    => 'US',
+                'telephone'     => '+1 123 111 3333',
+            ]
+        ];
+
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $result = $this->currentMock->prepareAddressData($postData);
+
+        $this->assertEquals($actual, $result);
+    }
+
+    public function testSaveActionWhereBoltReferenceIsFalse()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest', 'getResponse', '_redirect'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', false);
```

Comment:
I think in the real world, bolt_reference is undefined, not false

Reply:
fixed.